File: 02_Sliding_Window/02_minimum_size_subarray_sum/solution.py
# LeetCode 209: Minimum Size Subarray Sum
# https://leetcode.com/problems/minimum-size-subarray-sum/

class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        min_len = float('inf')
        left = 0
        window_sum = 0

        for right in range(len(nums)):
            window_sum += nums[right]

            while window_sum >= target:
                min_len = min(min_len, right - left + 1)
                window_sum -= nums[left]
                left += 1

        return 0 if min_len == float('inf') else min_len

File: 02_Sliding_Window/02_minimum_size_subarray_sum/solution.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int left = 0, currentSum = 0;
        int minLen = INT_MAX;

        for (int right = 0; right < nums.size(); right++){
            currentSum += nums[right];

            while (currentSum >= target){
                minLen = min(minLen, right - left + 1);
                currentSum -= nums[left];
                left++;
            }
        }
        return minLen == INT_MAX ? 0 : minLen;
    }
};